src/background.js

```diff
@@ -13,25 +13,38 @@ chrome.runtime.onInstalled.addListener(function () {
     });
 });
 
-function sendVideoUrlToMetube(videoUrl, metubeUrl, format, callback) {
+function sendVideoUrlToMetube(videoUrl, metubeUrl, format, advancedSettings, callback) {
     console.log("Sending videoUrl=" + videoUrl + " to metubeUrl=" + metubeUrl);
 
     if (typeof callback !== 'function') {
         callback = function () {
         };
     }
 
+    let {hostname} = new URL(videoUrl)
+
+    let postData = {
+      "quality": "best",
+      "format": format,
+      "url": videoUrl,
+      'auto_start': !advancedSettings['disable_auto_start'] ?? true
+    }
+
+
+    Object.keys(advancedSettings).forEach((key) => {
+      if (advancedSettings[key] && !['disable_auto_start'].includes(key) ) {
```

Comment:
it's need to be fixed here too.

![](https://storage.rpsl.info/s/screenshot-2024-03-23-19-23-dbbe7cd870.png
)

Reply:
uhm... i thing the error is still for `'auto_start': !advancedSettings['disable_auto_start'] ?? true`

Reply:
maybe something like `'auto_start':  'disable_auto_start' in advancedSettings ? !advancedSettings['disable_auto_start'] : true`  will fix it?

Reply:
Idk. I would try to make `advancedSettings` an object with properties and work with them. a little more code, but less abstraction

Reply:
looked one more time. i think `(advancedSettings && !advancedSettings['disable_auto_start']) ?? true` will fix that

Reply:
Sorry i was busy and totally forgot about this. I've pushed another change (did end up going with the default value approach. You where right it's the cleaner solution)